internal/plugin/servicegen.go

```diff
@@ -183,6 +183,12 @@ func (s serviceGenerator) generateMethodQuery(
 	}
 	// fields covered by path
 	pathCovered := make(map[string]struct{})
+	for _, segment := range rule.Template.Segments {
+		if segment.Kind != httprule.SegmentKindVariable {
+			continue
+		}
+		pathCovered[segment.Variable.FieldPath.String()] = struct{}{}
```

Comment:
Is this a variant of using a Set in other languages? E.g. We don't care about the map value, just that a key exists?

Reply:
Exactly! 

Typically you declare a `x := map[string]struct{}` as your "set".
Set values with `x[key] = struct{}{}`, and check values with `_, ok := x[key]`